src/rgw/driver/sfs/bucket.cc

```diff
@@ -385,14 +460,33 @@ std::unique_ptr<MultipartUpload> SFSBucket::get_multipart_upload(
       << "bucket::" << __func__ << ": oid: " << with_oid
       << ", upload id: " << with_upload_id << dendl;
 
-  std::string id = with_upload_id.value_or("");
-  if (id.empty()) {
-    id = bucket->gen_multipart_upload_id();
+  std::string next_oid(with_oid);
```

Comment:
Question.

What happens if, for whatever reason, `try_resolve_mp_from_oid` is not successful?
If my understanding is correct, it'll call `next_upload_id = with_upload_id.value_or("");`, but `next_oid` will be already assigned to the value of `with_oid`.

Maybe this is a corner case, but I'm just curious, because in that case we return a multipart with a value that maybe is not the one we were looking for.

What's confusing to me is that in the `try_resolve_mp_from_oid` function `next_oid` is returned with the value of the multipart `object_name`, but in this particular case it would be returned with the `oid` passed from `LC`.

Reply:
try_resolve_mp_from_oid may be unsuccessful if 
- something not LC called without an upload_id (I did not find any place, but that may change)
- LC called, but the MP is already gone

We can't 100% distinguish LC vs non-LC callers. The LC caller passes infos different enough from others, that we can just try to find a LC MP with try_resolve_mp_from_oid. In the LC case oid is a database id - in the non-LC case oid is an object_name. My read of the SAL / Op code is that 'oid' used to refer to a Ceph object id not a S3 object key. We are a bit lost in translation as our IDs are numbers and our names aren't part of the identifier space.

Basically the code does what we did before if we could not resolve to a LC MP